### DISCUSSION TITLE
Need a short help regarding access-expose

### DISCUSSION THREAD

**Discussion by Burki24** [Koenkk/zigbee2mqtt#9688] [Devices]
Hi all,

sorry, but I'm not familar to bitmasks and I searched on the guide to have a look what access-property: 3 means. But I didn't find that. there are some examples (like access: 7, 5, 1, 2)

Does anyone kno, what access: 3 means?

Many thanks in advance for your support.

Greetings from Berlin
Burkhard

**Comment by sjorge** [comment]
https://github.com/Koenkk/zigbee-herdsman-converters/blob/951921f877dc496eca2c078502a0a3414c1d1efd/lib/exposes.js#L421-L4409

They are defined here, a bit mask is just a set of bits where 0=not support, 1=support, for access the bit mask is made up of 3 bits.

Each unique base value has it's own bit and the combined values just set multiple of those to form the mask. Hopefully the table makes it a bit more clear.

Each bit also has a decimal value:
- bit0 = 1
- bit1 = 2
- bit3 = 4

So you can represent them by using an decimal too.

| property | bit2 | bit1 | bit 0 | | decimal |
| --- | --- | --- | --- | --- | -- | 
| *base values* | | | | | | |
| state | 0 | 0 | 1 | | 1 |
| set | 0 | 1 | 0 | | 2 |
| get | 1 | 0 | 0 | | 4 |
| *combined values* | | | | | |
| state_set | 0 | 1 | 1 | | 3 (1+2) |
| state_get | 1 | 0 | 1 | | 5 (1+4) |
| all | 1 | 1 | 1 | | 7 (1+2+4) |

Hope this helps a bit.

**Reply by Burki24** [reply]
many thanks. That's exactly what I needed.
